crates/ty_python_semantic/src/types/function.rs

```diff
@@ -198,6 +198,9 @@ pub struct OverloadLiteral<'db> {
     /// A set of special decorators that were applied to this function
     pub(crate) decorators: FunctionDecorators,
 
+    /// If `Some` then contains the message from @deprecated
+    pub(crate) deprecated: Option<ast::name::Name>,
```

Comment:
Use `returns(ref)` or `returns(deref)` to prevent that the accessor clones `Name` on every read
```suggestion
    /// If `Some` then contains the message from @deprecated
    #[returns(ref)]
    pub(crate) deprecated: Option<ast::name::Name>,
```